src/string/_internal/getWords.ts

```diff
@@ -16,6 +16,81 @@
  */
 const CASE_SPLIT_PATTERN = /[A-Z]?[a-z]+|[0-9]+|[A-Z]+(?![a-z])/g;
 
-export function getWords(str: string): string[] {
-  return Array.from(str.match(CASE_SPLIT_PATTERN) ?? []);
+export function getWords<T extends string>(str: T): CaseSplit<T> {
+  return Array.from(str.match(CASE_SPLIT_PATTERN) ?? []) as CaseSplit<T>;
```

Comment:
I think you can do this:

```suggestion
  return Array.from(str.match(CASE_SPLIT_PATTERN) ?? []) as any;
```

or :

```suggestion
  return Array.from(str.match(CASE_SPLIT_PATTERN) ?? []) as never;
```

to make it shorter and easier to replace (now you'll have a single place to change if you'll have a slightly different return type)

WDYT?